### ISSUE TITLE
kpsewhich with option --all

### ISSUE THREAD

**Issue by hvoss49** [MiKTeX/miktex#768]
- [ x] I have read and acknowledged this HOWTO article:  
  [Submitting a MiKTeX problem report](https://miktex.org/howto/bug-report)

```
Microsoft Windows [Version 10.0.18363.1440]
(c) 2019 Microsoft Corporation. All rights reserved.

C:\Users\Herbert Voss>kpsewhich --all pstricks.con
C:/Users/Herbert Voss/AppData/Local/Programs/MiKTeX/tex/generic/pstricks/pstricks.con
C:/Users/Herbert Voss/AppData/Local/Programs/MiKTeX/tex/generic/pstricks/pstricks.con
C:/Users/Herbert Voss/AppData/Local/Programs/MiKTeX/tex/xelatex/xetex-pstricks/pstricks.con
C:/Users/Herbert Voss/AppData/Local/Programs/MiKTeX/tex/xetex/xetex-pstricks/pstricks.con

```

Why do I get the first entry twice. And it happens only for the `tex/generic/` directories _and_
the option `--all`:

```
C:\Users\Herbert Voss>kpsewhich --all pstricks.sty
C:/Users/Herbert Voss/AppData/Local/Programs/MiKTeX/tex/latex/pstricks/pstricks.sty

C:\Users\Herbert Voss>kpsewhich --all pstricks.tex
C:/Users/Herbert Voss/AppData/Local/Programs/MiKTeX/tex/generic/pstricks/pstricks.tex
C:/Users/Herbert Voss/AppData/Local/Programs/MiKTeX/tex/generic/pstricks/pstricks.tex


C:\Users\Herbert Voss>kpsewhich pst-plot.tex
C:/Users/Herbert Voss/AppData/Local/Programs/MiKTeX/tex/generic/pst-plot/pst-plot.tex
```

No problem with TeXLive

```
ReportDate: 2021-03-29 11:51:18
CurrentVersion: 21.3
SetupDate: 2021-03-28 10:37:00
SetupVersion: 21.2
Configuration: Regular
GitInfo: ab8ebca / 2021-03-19 10:22:46
OS: Windows 10 Pro, 64-bit, build 18363
SharedSetup: no
LinkTargetDirectory: C:\Users\Herbert Voss\AppData\Local\Programs\MiKTeX\miktex\bin\x64
PathOkay: yes
LastUpdateCheck: 2021-03-28 10:48:06
LastUpdate: 2021-03-28 10:39:32
LastUpdateDb: 2021-03-28 10:47:53
SystemAdmin: yes
RootPrivileges: no
AdminMode: no
Root0: C:\Users\Herbert Voss\AppData\Roaming\MiKTeX
Root1: C:\Users\Herbert Voss\AppData\Local\MiKTeX
Root2: C:\Users\Herbert Voss\AppData\Local\Programs\MiKTeX
Root3: C:\ProgramData\MiKTeX
Root4: C:\Program Files\MiKTeX
UserInstall: C:\Users\Herbert Voss\AppData\Local\Programs\MiKTeX
UserConfig: C:\Users\Herbert Voss\AppData\Roaming\MiKTeX
UserData: C:\Users\Herbert Voss\AppData\Local\MiKTeX
CommonInstall: C:\Program Files\MiKTeX
CommonConfig: C:\ProgramData\MiKTeX
CommonData: C:\ProgramData\MiKTeX
```

**Comment by edocevoli** [comment]
Actually it happens also with `tex/latex` directories:

```sh
kpsewhich --progname pdflatex --all pstricks.sty
```

returns:

```
.../tex/latex/pstricks/pstricks.sty
.../tex/latex/pstricks/pstricks.sty
```

**Comment by edocevoli** [comment]
It happens with MiKTeX because the search path looks like this:

```
TEXINPUTS.pdflatex      = $TEXMFDOTDIR;$TEXMF/tex/{latex,generic,}//
```

This expands to:

```
.
$TEXMF/tex/latex//
$TEXMF/tex/generic//
$TEXMF/tex//
```

All `latex` and `generic` files are duplicated. TeX Live seems to filter out the duplicates.

**Comment by hvoss49** [comment]
A, I see. That explains it

**Comment by edocevoli** [comment]
Thank you, this has been fixed with v21.6.
